apps/aecore/src/aec_db.erl

```diff
@@ -609,7 +622,48 @@ get_genesis_hash() ->
     get_chain_state_value(genesis_hash).
 
 get_top_block_hash() ->
-    get_chain_state_value(top_block_hash).
+    case get_chain_state_value(top_block_node) of
+        #{hash := Hash} ->
+            Hash;
+        undefined ->
+            undefined
+    end.
+
+get_top_block_node() ->
+    get_chain_state_value(top_block_node).
+
+%% Some migration code: Ideally, top_block_node is there, and we're done.
+%% If not, we should find top_block_hash. Fetch the corresponding
+%% header, delete the obsolete top_block_hash and put in the new top_block_node.
+convert_top_block_entry() ->
+    ?t(convert_top_block_entry_()).
+
+convert_top_block_entry_() ->
+    case get_chain_state_value(top_block_node) of
+        undefined ->
+            case get_chain_state_value(top_block_hash) of
+                undefined ->
+                    weird;
```

Comment:
This will be the case for a fresh DB - the genesis block gets written in aec_conductor ;)

Reply:
Yes, I noted that too - not so weird :)